main.rb

```diff
@@ -0,0 +1,118 @@
+module Enumerable
+  
+  # my each
+  def my_each
+    count = 0
+    while count < length
+      yield(self[count])
+      count += 1
+    end
+    self
+  end
+  
+  # my_each_with_each
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+  
+  # my_select
+  def my_select
+    count = 0
+    arr = []
+    while count < length
+      arr.push(self[count]) if yield(self[count])
+      count += 1
+    end
+    arr
+  end
+  
+  # my_all
+  def my_all?
+    count = 0
+    while count < length
+      count += 1
+      return false unless yield(self[count])
+      
+    end
+    true
+  end
+  
+  # my_any?
+  def my_any?
+    count = 0
+    while count < length
+      count += 1
+      return true if yield(self[count])
+    end
+    false
+  end
+  
+  # my_none?
+  def my_none?
```

Comment:
We run into another problem here, it is not returning the proper value, the index is starting at 0 in this case so thats good. But with the same test case if I pass this block `[ 1, 1, 1, 1, 1] { |n| n == 1 }` it returns true, when it should return false. Do kindly see that it replicates the behavior of the `none?` ruby enumerable.